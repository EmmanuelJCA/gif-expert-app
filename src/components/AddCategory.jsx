import { useState } from "react"

const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('')

    const handleInputChange = ({ target }) => {
        setInputValue(target.value)
    }

    const handleSubmit = ( event ) => {
        event.preventDefault()

        if( inputValue.trim().length <= 1 ) return

        setInputValue('')
        onNewCategory( inputValue.trim() )
    }

    return (
        <form onSubmit={ handleSubmit }>
            <input 
            type="text" 
            placeholder="Buscar gif"
            value={ inputValue }
            onChange={ handleInputChange }
            />
        </form>
    )
}

export default AddCategory