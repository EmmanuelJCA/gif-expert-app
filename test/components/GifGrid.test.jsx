import { render, screen } from '@testing-library/react'
import { default as GifGrid } from '../../src/components/GifGrid'
import { useFetchGifs } from '../../src/hooks/useFetchGifs'

jest.mock('../../src/hooks/useFetchGifs')

describe('Tests on <GifGrid />', () => {

    const category = 'One Punch'

    test('should match the snapshot', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        })

        const { container } = render(
            <GifGrid category={ category } />
        )
        expect( container ).toMatchSnapshot()

    })

    test('should show the loading only', () => {

        useFetchGifs.mockReturnValue({
            images: [],
            isLoading: true,
        })

        render(
            <GifGrid category={ category } />
        )
        expect( screen.getByText('Cargando...') )
        expect( screen.getByText( category ) )

    })

    test('should show items when images are loaded via useFetchGifs', () => {

        const gifs = [
            {
                id: 'ABC',
                title: 'Saitama',
                url: 'https://gif/saitama.png'
            },
            {
                id: '123',
                title: 'Goku',
                url: 'https://gif/goku.png'
            },
        ]

        useFetchGifs.mockReturnValue({
            images: gifs,
            isLoading: true,
        })

        render(
            <GifGrid category={ category } />
        )

        expect( screen.getAllByRole('img').length ).toBe(2)

    })

})