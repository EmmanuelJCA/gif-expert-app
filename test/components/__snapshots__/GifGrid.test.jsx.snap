// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on <GifGrid /> should match the snapshot 1`] = `
<div>
  <h3>
    One Punch
  </h3>
  <h2>
    Cargando...
  </h2>
  <div
    class="card-grid"
  />
</div>
`;
